### PR TITLE
feat(iOS): open the Simulator app by default.

### PR DESCRIPTION
**Pull request by asafkorem**
As part of Detox v20 changes.

### CODE REVIEW

**Comment by d4vidi** [comment]
@asafkorem please rebase over wix:next so we could see the actual changes rather than many unrelated ones over 49 files 😆

**Comment by asafkorem** [comment]
> @asafkorem please rebase over wix:next so we could see the actual changes rather than many unrelated ones over 49 files 😆

The files and commits you see here are `next`'s changes, I was already above this branch, I guess that `next` was rebased or something so GitHub got a little confused. Rebasing again anyway 😄

**Comment by d4vidi** [comment]
> > @asafkorem please rebase over wix:next so we could see the actual changes rather than many unrelated ones over 49 files 😆
> 
> The files and commits you see here are `next`'s changes, it was already above this branch, I guess that `next` was rebased or something so GitHub got a little confused.. Rebasing again anyway 😄

That's legit 😄

File: detox/src/devices/common/drivers/ios/tools/AppleSimUtils.js [hunk 1]
```diff
@@ -72,6 +73,10 @@ class AppleSimUtils {
     return device;
   }
 
+  async _openSimulatorApp() {
+    await childProcess.execWithRetriesAndLogs(`open -a Simulator.app`, { retries: 1 });
```
Comment (noomorph) [inline detox/src/devices/common/drivers/ios/tools/AppleSimUtils.js:77]: Why not `retries: 0` ?

**Review by noomorph — CHANGES_REQUESTED** [review]
I'd suggest setting zero retries.

File: detox/src/devices/common/drivers/ios/tools/AppleSimUtils.js (hunk 1 above)
Comment (asafkorem) [reply detox/src/devices/common/drivers/ios/tools/AppleSimUtils.js:77]: Oh right, it's **re**tries, I believe I was thinking of total tries :|... Fixing. :+1:

**Comment by asafkorem** [comment]
Done @noomorph

**Comment by asafkorem** [comment]
Pushed another fix (add warning & use `--headless` on CI) @noomorph.

**Comment by noomorph** [comment]
I'll look tomorrow.

File: detox/index.d.ts [hunk 2]
```diff
@@ -43,12 +43,6 @@ declare global {
              * @example testRunner: 'mocha'
              */
             testRunner?: string;
-            /**
```
Comment (noomorph) [inline detox/index.d.ts:46]: Looks odd... like a bad merge, hmm.

Comment (noomorph) [inline detox/index.d.ts:46]: Looks odd... like a bad merge, hmm.

**Review by noomorph — APPROVED** [review]

File: detox/src/configuration/composeDeviceConfig.js [hunk 3]
```diff
@@ -195,23 +195,23 @@ function applyCLIOverrides(deviceConfig, cliConfig) {
   _assignCLIConfigIfSupported('readonly-emu', cliConfig.readonlyEmu, deviceConfig, 'readonly');
 }
 
-function _assignCLIConfigIfSupported(configName, configValue, deviceConfig, deviceConfigKey) {
-  if (!configValue) {
+function _assignCLIConfigIfSupported(argName, argValue, deviceConfig, propertyName) {
+  if (argValue === undefined) {
```
Comment (asafkorem) [inline detox/src/configuration/composeDeviceConfig.js:199]: :+1:

File: detox/src/configuration/composeDeviceConfig.test.js [hunk 4]
```diff
@@ -370,9 +370,9 @@ describe('composeDeviceConfig', () => {
             beforeEach(() => setConfig(deviceType, configType));
 
             it('should override .headless without warnings', () => {
-              cliConfig.headless = true;
+              cliConfig.headless = false;
               expect(compose()).toEqual(expect.objectContaining({
-                headless: true,
+                headless: false,
```
Comment (asafkorem) [inline detox/src/configuration/composeDeviceConfig.test.js:375]: Why did you change that? 🤔

Comment (asafkorem) [reply detox/src/configuration/composeDeviceConfig.test.js:375]: OK got that now, because the `!value` thing, great catch :+1:
